packages/stokhos/CMakeLists.txt

```diff
@@ -10,6 +10,14 @@ TRIBITS_PACKAGE(Stokhos)
 
 TRIBITS_ADD_EXPLICIT_INSTANTIATION_OPTION()
 
+SET (Stokhos_ENABLE_DEPRECATED_CODE_DEFAULT ON)
+TRIBITS_ADD_OPTION_AND_DEFINE(
+  Stokhos_ENABLE_DEPRECATED_CODE
+  STOKHOS_ENABLE_DEPRECATED_CODE
+  "Whether Stokhos enables deprecated code (that is, anything marked with the STOKHOS_DEPRECATED macro) at compile time.  Default is ON (deprecated code enabled)."
+  ${Stokhos_ENABLE_DEPRECATED_CODE_DEFAULT}
+)
+
```

Comment:
The tribits system has a deprecation macro for packages to call so that we can toggle deprecated code on and off at the trilinos level instead of having to toggle for each package independently. You might want to add this into this PR.

function is:
TRIBITS_ADD_SHOW_DEPRECATED_WARNINGS_OPTION()

found in:
Trilinos/cmake/tribits/core/package_arch/TribitsPackageMacros.cmake

also wanted to mention this file for including some convenient macros to pull into a package's config file for deprecation markings:
Trilinos/cmake/tribits/core/package_arch/TribitsConfigureFile.cmake

Reply:
I saw that, but it wasn't clear to me what it actually did.  I modeled this off of what is in Tpetra:
```
TRIBITS_ADD_SHOW_DEPRECATED_WARNINGS_OPTION()

TRIBITS_SET_ST_FOR_DEV_MODE(ST_FOR_DEV_PT_FOR_RELEASE)

SET (Tpetra_ENABLE_DEPRECATED_CODE_DEFAULT ON)
TRIBITS_ADD_OPTION_AND_DEFINE(
  Tpetra_ENABLE_DEPRECATED_CODE
  TPETRA_ENABLE_DEPRECATED_CODE
  "Whether Tpetra enables deprecated code (that is, anything marked with the TPETRA_DEPRECATED macro) at compile time.  Default is ON (deprecated code enabled)."
  ${Tpetra_ENABLE_DEPRECATED_CODE_DEFAULT}
)
```
which calls that macro, but also has its own variable for `Tpetra_ENABLE_DEPRECATED_CODE_DEFAULT`.  Is that not the correct way to do this?  Are we supposed to be using `{${PROJECT_NAME}_HIDE_DEPRECATED_CODE` instead?

Reply:
Just so I don't say something that may be outdated - @sebrowne and @bartlettroscoe what's the current procedure for handling deprecated code?

Reply:
@etphipp and @rppawlo, handling of deprecated code is built into TriBITS. See:

* https://tribitspub.github.io/TriBITS/users_guide/index.html#regulated-backward-compatibility-and-deprecated-code

Reply:
Thanks.  I was looking for something like this originally, but didn't find it, so I just tried to model it off of an existing package.  I guess I chose the wrong package!